File: src/main/java/org/bitecodelabs/botlerdaemon/events/ServerStartHandler.java
package org.bitecodelabs.botlerdaemon.events;

import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents;
import net.minecraft.server.MinecraftServer;
import org.bitecodelabs.botlerdaemon.Daemon;
import org.bitecodelabs.botlerdaemon.connections.SocketClient;

public class ServerStartHandler implements ServerLifecycleEvents.ServerStarted {
    @Override
    public void onServerStarted(MinecraftServer server) {
        SocketClient.getInstance(server);

        Daemon.LOGGER.info("Botler Daemon has started");
    }
}

File: src/main/kotlin/org/bitecodelabs/botlerdaemon/events/ServerStartHandler.kt
package org.bitecodelabs.botlerdaemon.events

import net.fabricmc.fabric.api.event.lifecycle.v1.ServerLifecycleEvents
import net.minecraft.server.MinecraftServer
import org.bitecodelabs.botlerdaemon.Daemon
import org.bitecodelabs.botlerdaemon.connections.SocketClient

class ServerStartHandler: ServerLifecycleEvents.ServerStarted {
    override fun onServerStarted(server: MinecraftServer) {

        SocketClient.getInstance(server)

        Daemon.LOGGER.info("Socket client initialized")

        Daemon.LOGGER.info("Botler Daemon is started")
    }

}